jobs/build-charms/builder_launchpad.py

```diff
@@ -58,15 +58,23 @@ def __init__(self, *args, **kwargs) -> None:
     @cached_property
     def _lp(self):
         """Use launchpad credentials to interact with launchpad."""
-        creds = os.environ.get("LPCREDS", None)
-        parser = ConfigParser()
-        parser.read(creds)
-        return Launchpad.login_with(
-            application_name=parser["1"]["consumer_key"],
-            service_root="production",
-            version="devel",
-            credentials_file=creds,
-        )
+        cred_file = os.environ.get("LPCREDS", None)
+        creds_local = os.environ.get("LPLOCAL", None)
+        if cred_file:
+            parser = ConfigParser()
+            parser.read(cred_file)
+            return Launchpad.login_with(
+                application_name=parser["1"]["consumer_key"],
+                service_root="production",
+                version="devel",
+                credentials_file=cred_file,
+            )
+        elif creds_local:
+            return Launchpad.login_with(
+                "localhost",
+                "production",
+                version="devel",
+            )
```

Comment:
facilitates a way to run the job locally if no LPCREDS credentials file is available

the runner will face a URL prompt to gain an OAUTH token